src/Core/Indexes/QuadTree.cs

```diff
@@ -0,0 +1,400 @@
+﻿namespace AEGIS.Indexes
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Linq;
+
+    public class QuadTree : ISpatialIndex
+    {
+        protected class QuadTreeNode
+        {
+            /// <summary>
+            /// The region of the node.
+            /// </summary>
+            private Envelope bound;
+
+            /// <summary>
+            /// The children nodes of this node. Each internal node must always have 4 children.
+            /// </summary>
+            private List<QuadTreeNode> children = new List<QuadTreeNode>(4);
+
+            /// <summary>
+            /// The geometries stored in this node.
+            /// </summary>
+            private List<IBasicGeometry> contents = new List<IBasicGeometry>();
+
+            /// <summary>
+            /// Initializes a new instance of the <see cref="QuadTreeNode"/> class.
+            /// </summary>
+            /// <param name="bound">The region of the node.</param>
+            public QuadTreeNode(Envelope bound)
+            {
+                this.bound = bound;
+            }
+
+            /// <summary>
+            /// Gets the region of the node.
+            /// </summary>
+            /// <value> The region of the node.</value>
+            public Envelope Envelope
+            {
+                get
+                {
+                    return this.bound;
+                }
+            }
+
+            /// <summary>
+            /// Gets a value indicating whether the node is a leaf node.
+            /// </summary>
+            /// <value><c>true</c> if the node is a leaf node, otherwise <c>false</c>.</value>
+            public Boolean IsLeaf { get { return this.children.Count == 0; } }
+
+            /// <summary>
+            /// Gets a value indicating whether the node is empty or not.
+            /// </summary>
+            /// <value><c>true</c> if the node is a leaf node and has no geometries, otherwise <c>false</c>.</value>
+            public Boolean IsEmpty
+            {
+                get
+                {
+                    return this.contents.Count == 0 && this.IsLeaf;
+                }
+            }
+
+            /// <summary>
+            /// Gets the contents of this node.
+            /// </summary>
+            /// <value> The geometries stored in this node.</value>
```

Comment:
Please check for spaces before and after comment, as they are taken up by VS. I.e. it should be "<value>The geometries ".